handlers/officerDeltaHandler.go

```diff
@@ -1,22 +1,45 @@
 package handlers
 
 import (
+	"github.com/companieshouse/chs-delta-api/helpers"
+	"github.com/companieshouse/chs-delta-api/services"
 	"net/http"
 )
 
-// OfficerDeltaHandler offers a handler by which to publish a message onto a kafka topic.
+const (
+	officersTopic = "officers-delta"
+)
+
+// OfficerDeltaHandler offers a handler by which to publish an office-delta onto the officer-delta kafka topic.
 type OfficerDeltaHandler struct {
+	KSvc services.KafkaService
+	h helpers.Helper
 }
 
 // NewOfficerDeltaHandler returns an OfficerDeltaHandler.
-func NewOfficerDeltaHandler() *OfficerDeltaHandler {
-	return &OfficerDeltaHandler{}
+func NewOfficerDeltaHandler(kSvc services.KafkaService, h helpers.Helper) *OfficerDeltaHandler {
+	return &OfficerDeltaHandler{KSvc: kSvc, h: h}
 }
 
 // ServeHTTP accepts an incoming OfficerDelta request via a POST method, validates it
-// and then passes it to a Kafka service for further processing. If errors are encountered
-// then they will be returned via the ResponseWriter.
-func (kp *OfficerDeltaHandler) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
+// and then passes it to a Kafka service for further processing along with an officers-delta topic. If errors are
+// encountered then they will be returned via the ResponseWriter.
+func (kp *OfficerDeltaHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	// Get request body.
+	data, err := kp.h.GetDataFromRequest(r)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("Error retrieving request body: " + err.Error())) // TODO: Temp until we add the CH errors object
```

Comment:
I don't think we should be returning the error to the user so that we can avoid giving them knowledge of our system.

Instead, we should log the error, and give a generic response like: 

```
if err != nil {
			log.ErrorR(req, err)
			m := models.NewMessageResponse("failed to validate the practitioner request supplied")
			utils.WriteJSONWithStatus(w, req, m, http.StatusInternalServerError)
			return
		}
```

Reply:
I will remove the returning of the error directly, we plan on adding validation and properly formatting the response to the caller later in other tickets (DSND-87)